### PR TITLE
[Refactor] Separate Sounds

### PR DESCRIPTION
**Pull request by bjornstar**
I noticed on a low end system the horn could get quite delayed. This PR attempts to improve responsiveness of the sound effects by separating them into individual components. The trade-off is more boilerplate.

- `useToggle` now accepts an array and all entries must be true to toggle it on
- Prefer `useToggle` for all instances of audio to prevent large component re-renders (the train would noticeably flicker when toggling sound)
- `BoostAudio` stops when we run out of boost
- `BrakeAudio` stops when we slow down enough
- `rpmTarget` is now on the mutation object and is calculated alongside speed
- Fixed a bug where the horn and brake could get stuck on when turning off sounds

### CODE REVIEW

File: src/models/vehicle/Chassis.tsx [hunk 1]
```diff
@@ -60,57 +62,56 @@ interface ChassisGLTF extends GLTF {
...
 
+type MaterialMesh = Mesh<BoxBufferGeometry, MeshStandardMaterial>
+
+const gears = 10
 const c = new Color()
 const v = new Vector3()
 
-/*
-Auto-generated by: https://github.com/pmndrs/gltfjsx
-author: Alexus16 (https://sketchfab.com/Alexus16)
```
Comment (njm222) [inline src/models/vehicle/Chassis.tsx:68]: I think we should keep the attribution

File: src/effects/audio/Accelerate.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import type { PositionalAudio as PositionalAudioImpl } from 'three'
+
+import { mutation, useStore } from '../../store'
+
+const { lerp } = MathUtils
+
+export const AccelerateAudio = () => {
+  const ref = useRef<PositionalAudioImpl>(null)
+  const [maxSpeed] = useStore(({ vehicleConfig: { maxSpeed } }) => [maxSpeed])
```
Comment (njm222) [inline src/effects/audio/Accelerate.tsx:14]:
```suggestion
  const maxSpeed = useStore(({ vehicleConfig: { maxSpeed } }) => maxSpeed)
```

File: src/effects/audio/Engine.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import type { PositionalAudio as PositionalAudioImpl } from 'three'
+
+import { mutation, useStore } from '../../store'
+
+const { lerp } = MathUtils
+
+export const EngineAudio = () => {
+  const ref = useRef<PositionalAudioImpl>(null)
+  const [maxSpeed] = useStore(({ vehicleConfig: { maxSpeed } }) => [maxSpeed])
```
Comment (njm222) [inline src/effects/audio/Engine.tsx:14]:
```suggestion
  const maxSpeed = useStore(({ vehicleConfig: { maxSpeed } }) => maxSpeed)
```

File: src/models/vehicle/Vehicle.tsx [hunk 4]
```diff
@@ -1,26 +1,23 @@
...
 import { Wheel } from './Wheel'
 
 import type { Controls, WheelInfo } from '../../store'
-import type { ChassisProps } from './Chassis'
 
 const { lerp } = MathUtils
 const v = new Vector3()
 
-type VehicleProps = PropsWithChildren<Pick<ChassisProps, 'angularVelocity' | 'position' | 'rotation'>>
+type VehicleProps = PropsWithChildren<Pick<BoxProps, 'angularVelocity' | 'position' | 'rotation'>>
```
Comment (njm222) [inline src/models/vehicle/Vehicle.tsx:20]: Not sure why this was changed?

**Review by njm222 — CHANGES_REQUESTED** [review]
This looks way better, thanks!

File: src/models/vehicle/Chassis.tsx (hunk 1 above)
Comment (bjornstar) [reply src/models/vehicle/Chassis.tsx:68]: It was duplicated.

File: src/models/vehicle/Vehicle.tsx (hunk 4 above)
Comment (bjornstar) [reply src/models/vehicle/Vehicle.tsx:20]: ChassisProps was just BoxProps, so no need for it.

**Review by njm222 — APPROVED** [review]

**Pull request merged**